File: python-sdk/src/liveblocks/models/shared/createauthorization.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from liveblocks import utils


@dataclass_json
@dataclasses.dataclass
class CreateAuthorization:
    group_ids: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('groupIds') }})
    user_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userId') }})
    user_info: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userInfo') }})
    

File: typescript-sdk/src/sdk/models/shared/createauthorization.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class CreateAuthorization extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=groupIds" })
  groupIds?: string[];

  @SpeakeasyMetadata({ data: "json, name=userId" })
  userId?: string;

  @SpeakeasyMetadata({ data: "json, name=userInfo" })
  userInfo?: Record<string, any>;
}
